c_api/tests/c_api_aot_test.cpp

```diff
@@ -11,51 +11,25 @@ static void kernel_aot_test(TiArch arch) {
   std::stringstream aot_mod_ss;
   aot_mod_ss << folder_dir;
 
-  TiMemoryAllocateInfo alloc_info;
-  alloc_info.size = kArrLen * sizeof(int32_t);
-  alloc_info.host_write = false;
-  alloc_info.host_read = true;  // Enable for testing only
-  alloc_info.export_sharing = false;
-  alloc_info.usage = TiMemoryUsageFlagBits::TI_MEMORY_USAGE_STORAGE_BIT;
+  ti::Runtime runtime(arch);
 
-  TiRuntime runtime = ti_create_runtime(arch);
+  ti::NdArray<int32_t> arg1_array =
+      runtime.allocate_ndarray<int32_t>({kArrLen}, {1}, true);
+  ti::AotModule aot_mod = runtime.load_aot_module(aot_mod_ss.str().c_str());
+  ti::Kernel k_run = aot_mod.get_kernel("run");
 
-  // Load Aot and Kernel
-  TiAotModule aot_mod = ti_load_aot_module(runtime, aot_mod_ss.str().c_str());
-  TiKernel k_run = ti_get_aot_module_kernel(aot_mod, "run");
-
-  // Prepare Arguments
-  TiArgument arg0 = {.type = TiArgumentType::TI_ARGUMENT_TYPE_I32,
-                     .value = {.i32 = arg0_val}};
-
-  TiMemory memory = ti_allocate_memory(runtime, &alloc_info);
-  TiNdArray arg_array = {.memory = memory,
-                         .shape = {.dim_count = 1, .dims = {kArrLen}},
-                         .elem_shape = {.dim_count = 1, .dims = {1}},
-                         .elem_type = TiDataType::TI_DATA_TYPE_I32};
-
-  TiArgumentValue arg_value = {.ndarray = std::move(arg_array)};
-
-  TiArgument arg1 = {.type = TiArgumentType::TI_ARGUMENT_TYPE_NDARRAY,
-                     .value = std::move(arg_value)};
-
-  // Kernel Execution
-  constexpr uint32_t arg_count = 2;
-  TiArgument args[arg_count] = {std::move(arg0), std::move(arg1)};
-
-  ti_launch_kernel(runtime, k_run, arg_count, &args[0]);
-  ti_wait(runtime);
+  k_run[0] = arg0_val;
+  k_run[1] = arg1_array;
+  k_run.launch();
```

Comment:
nit: do we need to keep the ti_wait (or similar sync) here?

Reply:
Added